lib/src/server/server.dart

```diff
@@ -93,20 +93,33 @@ class Server {
       bool v6Only: false,
       bool shared: false}) async {
     // TODO(dart-lang/grpc-dart#9): Handle HTTP/1.1 upgrade to h2c, if allowed.
+    bool unix = false;
+    dynamic _address = address;
+    int _port = port;
+    if (address != null) {
+      // `address` which start with 'unix://' prefix, will be parsed as unix socket.
+      RegExp re = RegExp(r'(^unix://)');
+      Match m = re.firstMatch(address);
```

Comment:
This assumes that `address` is a string. If it's already an `InternetAddress`, then this code will throw.

So, the test above should be:
```dart
if (address is String) {
```

Reply:
Thanks for your suggests, but I have update the code as @mraleph suggested.

Reply:
Good catch @lrhn - though this code will go away if we do what https://github.com/grpc/grpc-dart/pull/327#discussion_r494137817 suggests (same applies to your other comments below, the parsing code would disappear).